chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/execute.rs

```diff
@@ -499,18 +499,15 @@ impl<'a> ExecuteReportContextRemainingAccountsLayout<'a> {
         self.messaging_accounts.is_some()
     }
 
-    pub fn receiver_and_message_account_keys(&self) -> impl Iterator<Item = Pubkey> {
-        let keys: Vec<Pubkey> = self
-            .messaging_accounts
+    pub fn receiver_and_message_account_keys(&self) -> Vec<Pubkey> {
+        self.messaging_accounts
             .iter()
             .flat_map(|accs| {
                 Some(accs.logic_receiver.key())
                     .into_iter()
                     .chain(accs.remaining_messaging_accounts.iter().map(|a| a.key()))
             })
-            .collect();
-
-        keys.into_iter()
+            .collect()
```

Comment:
The old function isn't quite right, but I'm not 100% on the new one either. Why not:

```rust
    pub fn receiver_and_message_account_keys(&self) -> impl Iterator<Item = Pubkey> + 'a {
        self.messaging_accounts.iter().flat_map(|accs| {
            Some(accs.logic_receiver.key())
                .into_iter()
                .chain(accs.remaining_messaging_accounts.iter().map(|a| a.key()))
        })
    }
```

Always best to return an existential iterator instead of a Vec in case the caller doesn't want to use heap memory.